packages/grafana-prometheus/src/datasource.ts

```diff
@@ -537,7 +537,15 @@ export class PrometheusDatasource
     const annotation = options.annotation;
     const { tagKeys = '', titleFormat = '', textFormat = '' } = annotation;
 
-    const step = rangeUtil.intervalToSeconds(annotation.step || ANNOTATION_QUERY_STEP_DEFAULT) * 1000;
+    const scopedVars = {
+      ...this.getIntervalVars(),
+      ...this.getRangeScopedVars(options?.range ?? getDefaultTimeRange()),
+    };
+
+    const step = rangeUtil.intervalToSeconds(
+      this.interpolateString(annotation.step || ANNOTATION_QUERY_STEP_DEFAULT, scopedVars)
+    ) * 1000;
+
```

Comment:
```suggestion
    const input = frames[0].meta?.executedQueryString || '';
    const regex = /Step:\s*([\d\w]+)/;
    const match = input.match(regex);
    const stepValue = match ? match[1] : null;
    const step = rangeUtil.intervalToSeconds(
      this.interpolateString(stepValue || ANNOTATION_QUERY_STEP_DEFAULT, scopedVars)
    ) * 1000;
```

Reply:
I do not want to get too excited about this suggestion, but it handled the annotation breaking apart on larger query ranges. Would you mind testing it out?